csharp/Adapter/Microsoft.Spark.CSharp/Network/RioNative.cs

```diff
@@ -128,25 +129,34 @@ private void Init()
             var maxThreads = 1;
             if (useThreadPool)
             {
-                maxThreads = Environment.ProcessorCount;
+                maxThreads = (Environment.ProcessorCount + 1) >> 1;
```

Comment:
These code is only for Multithread Worker. In simple worker, there are only two threads (separately for send and receive completion). No difference between local and yarn cluster mode.